core/application/impl/config_reader/pt_util.cpp

```diff
@@ -5,130 +5,18 @@
-      OUTCOME_TRY(genesis, scale::decode<primitives::Block>(genesis_bytes));
-      config.genesis = std::move(genesis);
-    } else if (!update) {
-      return ConfigReaderError::MISSING_ENTRY;
-    }
 
-    auto extrinsic_api_port_opt =
-        tree.get_child("api_ports").get_optional<uint16_t>("extrinsic");
-    if (extrinsic_api_port_opt) {
-      config.api_ports.extrinsic_api_port = extrinsic_api_port_opt.value();
-    } else if (!update) {
-      return ConfigReaderError::MISSING_ENTRY;
-    }
-
-    OUTCOME_TRY(pt_unwrap(
-        tree.get_child_optional("authorities"),
-        update,
-        [&config](const pt::ptree &node) -> outcome::result<void> {
-          return pt_foreach(
-              node,
-              [&config](const pt::ptree &authority) -> outcome::result<void> {
-                OUTCOME_TRY(bytes, common::unhex(authority.data()));
-                crypto::ED25519PublicKey pkey;
-                std::copy(bytes.begin(), bytes.end(), pkey.begin());
-                config.authorities.push_back(pkey);
-                return outcome::success();
-              });
-        }));
-
-    OUTCOME_TRY(pt_unwrap(
-        tree.get_child_optional("session_keys"),
-        update,
-        [&config](const pt::ptree &node) -> outcome::result<void> {
-          return pt_foreach(
-              node,
-              [&config](const pt::ptree &session_key) -> outcome::result<void> {
-                OUTCOME_TRY(bytes, common::unhex(session_key.data()));
-                crypto::SR25519PublicKey pkey;
-                std::copy(bytes.begin(), bytes.end(), pkey.begin());
-                config.session_keys.push_back(pkey);
-                return outcome::success();
-              });
-        }));
-
-    OUTCOME_TRY(pt_unwrap(
-        tree.get_child_optional("peers_info"),
-        update,
-        [&config](const pt::ptree &node) -> outcome::result<void> {
-          return pt_foreach(
-              node,
-              [&config](
-                  const pt::ptree &peer_info_data) -> outcome::result<void> {
-                auto peer_id = peer_info_data.get_child("id");
-                OUTCOME_TRY(id, PeerId::fromBase58(peer_id.data()));
-                PeerInfo peer_info{id, {}};
-                for (auto &[_, address_data] :
-                     peer_info_data.get_child("addresses")) {
-                  OUTCOME_TRY(address,
-                              Multiaddress::create(address_data.data()));
-                  peer_info.addresses.push_back(address);
-                }
-                config.peers_info.push_back(peer_info);
-                return outcome::success();
-              });
-        }));
-    if (update) {
-      conf.value() = config;
-    }
-    return config;
-  }
-
-  outcome::result<KagomeConfig> initConfigFromPropertyTree(
-      const pt::ptree &tree) {
-    return processConfigFromPropertyTree(tree, boost::none);
-  }
+  outcome::result<std::vector<uint8_t>> unhexWith0x(
```

Comment:
maybe we better change the unhex method to work with "0x" prefix as well as without it?

Reply:
I dont think so. This would increase complexity of the simple method. Unhex with 0x is not needed that frequently, so having this function in some util file is okay in my opinion